### PR TITLE
added labels to plots and handled Deprecation warnings

### PR DESCRIPTION
**Pull request by tarjebargheer**
For pitching persistent homology to some clients, I thought it would look better with labels on some of the plots. Feel free to use if desired - I have also modified the code to handle deprecation warnings of matplotlib (specified handles on the axis objects that are used instead of plt directly)

### CODE REVIEW

**Comment by mrzv** [comment]
You added `build/bindings/python/dionysus/plot.py`. I think you want it without `build`. Please fix that, and I'll look at the actual diff.

**Comment by mrzv** [comment]
BTW, you can add labels with the existing code as follows:

```
import dionysus as d
import matplotlib.pyplot as plt

d.plot.plot_diagram(dgms[1])

plt.axes().set_xlabel("birth")
plt.axes().set_ylabel("death")

plt.show()
```

**Comment by tarjebargheer** [comment]
When I do that on the existing code using matplotlib 2.1.2, I get the following Deprecation Warning:

> 
>    MatplotlibDeprecationWarning: Adding an axes using the same arguments as a previous axes currently reuses the earlier instance.  In a future version, a new instance will always be created and returned.  Meanwhile, this warning can be suppressed, and the future behavior ensured, by passing a unique label to each axes instance.
>    warnings.warn(message, mplDeprecation, stacklevel=1)

so I modified the code with the plotting routines running on the specifix axis in matplotlib - which might not be necessary before these matplotlib changes are implemented... staying close to your original code is fine with me... I will remove the build/ pull request as well

File: bindings/python/dionysus/plot.py [hunk 1]
```diff
@@ -9,14 +9,19 @@ def plot_diagram(dgm, show = False):
...
     max_death = max(p.death for p in dgm if p.death != inf)
 
-    plt.axes().set_aspect('equal', 'datalim')
+    ax = plt.axes()
+    ax.set_aspect('equal', 'datalim')
 
     min_diag = min(min_birth, min_death)
     max_diag = max(max_birth, max_death)
 
     plt.scatter([p.birth for p in dgm], [p.death for p in dgm])
```
Comment (mrzv) [inline bindings/python/dionysus/plot.py]: Should these not be done on `ax` as well? Both `scatter` and `plot`?

File: bindings/python/dionysus/plot.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-def plot_diagram(dgm, show = False):
+def plot_diagram(dgm, show = False,labels = False):
```
Comment (mrzv) [inline bindings/python/dionysus/plot.py]: Can you add a space after the comma, for consistency?

File: bindings/python/dionysus/plot.py [hunk 3]
```diff
@@ -59,17 +64,23 @@ def plot_diagram_density(dgm, bins = 200, lognorm = True, diagonal = True, show
...
-        plt.plot([min_birth, max_death], [min_birth, max_death])        # diagonal
+        ax.plot([min_birth, max_death], [min_birth, max_death])        # diagonal
 
     ## clip the view
     #plt.axes().set_xlim([min_birth, max_birth])
     #plt.axes().set_ylim([min_death, max_death])
-
-    plt.colorbar()
+    if labels:
+        plt.colorbar(im,label = "overlap quantity")
```
Comment (mrzv) [inline bindings/python/dionysus/plot.py]: Should `colorbar` be called on `ax`?

Comment (mrzv) [reply bindings/python/dionysus/plot.py]: Hm, actually, there is not `colorbar` in `ax`. Matplotlib is confusing.

**Comment by tarjebargheer** [comment]
Thanks, I have tried to follow your requests - and it should probably be better at handling the upcoming deprecation. apparently, colorbar takes the axes through an ax parameter and lives on the figure.

**Pull request merged**